### PR TITLE
IAM Identity: account settings

### PR DESCRIPTION
**Pull request by pooraniss-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by pooraniss-ibm** [comment]
Inside go.mod and go.sum files I have used the latest version of [platform-go-sdk version v0.31.0](https://github.com/IBM/platform-services-go-sdk/releases/tag/v0.31.0) 

if an older version is being used then the resourcecontroller is failing. this needs to be fixed by the respective contributor.

I have all the changes added for iamidentity resources and datasources.

**Comment by pooraniss-ibm** [comment]
Screenshot 2022-12-06 at 11 43 04

**Comment by pooraniss-ibm** [comment]
@hkantare rebased and updated, can be reviewed and merged

File: ibm/service/iamidentity/resource_ibm_iam_account_settings.go [hunk 1]
```diff
@@ -306,7 +357,10 @@ func resourceIbmIamAccountSettingsUpdate(context context.Context, d *schema.Reso
 		updateAccountSettingsOptions.SetMfa(mfa_str)
 		hasChange = true
 	}
-
+	if d.HasChange("user_mfa") {
+		// TODO: handle UserMfa of type TypeList -- not primitive, not model
```
Comment (hkantare) [inline ibm/service/iamidentity/resource_ibm_iam_account_settings.go:361]: We need to handle the update of "user_mfa"

File: ibm/service/iamidentity/resource_ibm_iam_account_settings_test.go [hunk 2]
```diff
@@ -126,9 +128,15 @@ func testAccCheckIbmIamAccountSettingsUpdateConfig() string {
 			restrict_create_platform_apikey = "%s"
 			if_match = "%s"
 			mfa = "%s"
+			user_mfa {
+				iam_id = "iam_id"
```
Comment (hkantare) [inline ibm/service/iamidentity/resource_ibm_iam_account_settings_test.go:132]: can we add a test case to add multiple users and update operation to remove or add users

**Comment by pooraniss-ibm** [comment]
@hkantare Hii, the comments are addressed and PR is updated :)

File: ibm/service/iamidentity/resource_ibm_iam_account_settings_test.go (hunk 2 above)
Comment (pooraniss-ibm) [reply ibm/service/iamidentity/resource_ibm_iam_account_settings_test.go:134]: I have added two tests, one is to test with an empty list of user_mfa, and another one with 2 users.

**Pull request merged**